Update plugin.py

hansvanlin opened the pull request:
Changed switch condition and added extra logging (needs cleanup)

hansvanlin committed:
Update plugin.py

Changed switch condition

tuk90 commented on plugin.py:
```diff
@@ -304,14 +314,19 @@ def update_devices(self, access_token):
                 value_firePlaceModeActive = response_json["appliances"][0]["climateZones"][0]["firePlaceModeActive"]
                 if value_firePlaceModeActive == True:
                     Devices[13].Update(nValue=1, sValue="On")
+                    Domoticz.Log("device 13 true = "+ str(value_firePlaceModeActive))
```
Device 13 does not say anything, can you make it the default as the other devices?

tuk90 commented on plugin.py:
```diff
@@ -304,14 +314,19 @@ def update_devices(self, access_token):
             else:
-                Devices[13].Update(nValue=0, sValue="Off")
+                Devices[13].Update(nValue=None, sValue=None)
+                Domoticz.Log("device 13 else")
```
remove or improve logging

tuk90 commented on plugin.py:
```diff
@@ -530,32 +545,37 @@ def zonemode(self, access_token, level):
-                Domoticz.Log("in the if loop van level 0")
+            if str(value_firePlaceModeActive) == "True": # Fireplace mode currently on
+                Domoticz.Log("value_firePlaceModeActive in loop is:" + str(value_firePlaceModeActive))
+                Domoticz.Log("in the if loop van true ---> False")
```
Same here

tuk90 commented on plugin.py:
```diff
@@ -530,32 +545,37 @@ def zonemode(self, access_token, level):
                 response.raise_for_status()
-                Domoticz.Log("Fireplace succesfully set to false")
+                Devices[13].Update(nValue=1, sValue="On")
+                Domoticz.Log("Fireplace succesfully set to true")
```
Logging should be: Fireplace Mode succesfully set to true

tuk90 commented on plugin.py:
```diff
@@ -530,32 +545,37 @@ def zonemode(self, access_token, level):
                     json=json_data
                     )
                 response.raise_for_status()
-                Domoticz.Log("Fireplace succesfully set to false")
```
Logging should be: Fireplace Mode succesfully set to false

tuk90 commented on plugin.py:
```diff
@@ -530,32 +545,37 @@ def zonemode(self, access_token, level):
         try:
-            if level == 0: # Fireplace mode on
-                Domoticz.Log("in the if loop van level 0")
+            if str(value_firePlaceModeActive) == "True": # Fireplace mode currently on
```
I am missing a catch for as the Remeha device does not have the capability for fireplace mode, this needs to be generic, you can take a look at what I did for the retrieving of the values, so add clause for the capability otherwise people might get strange errors

tuk90 commented on plugin.py:
```diff
@@ -530,32 +545,37 @@ def zonemode(self, access_token, level):
-                json_data = {"fireplaceModeActive": False}
+                Devices[13].Update(nValue=0, sValue="Off")
+                Domoticz.Log("Fireplace succesfully set to false")
+            elif str(value_firePlaceModeActive): # Fireplace mode currently off
```
Same here, build an extra argument in the elif clause

tuk90 reviewed (changes requested):
Hi,

I gave some comments.

hansvanlin committed:
Update plugin.py

removed debug logging
made device 13 default

hansvanlin committed:
Update plugin.py

typo

hansvanlin committed:
Update plugin.py

changed retrieving value_firePlaceModeActive

hansvanlin commented:
Hi Nick,
Thank you for all the comments, it's a fast way to learn python.
I made all the requested changes and I will let you know if I didn't break the fireplace Mode switch,
after I tested it.

tuk90 commented:
Hi,

I am not a software developer, I only have my boerenverstand :). With that in mind:

Thanks for the updates, I do not think it will work since you when calling the function your are submitting the level and in the function you are submitting a different value, even if it works it does not make any sense. Furthermore are you setting the global value in the function which seems quite odd since then if you click on the fireplace it will set the global vlaue each time? That seems a bit exagerating. Can you not just use the level of the switch to decide on what should be done?

hansvanlin committed:
Update plugin.py

removed the breaking part.
value_firePlaceModeActive is already defined in the readback part and then available as global with error handling.

hansvanlin commented:
Hi Nick, indeed it didn't work...I removed the part that I added.
The last file I checked already that I can switch the mode correctly
I update the button so that it has already the correct status when the api call is succeeded if it doesn't receive the updated value from remeha the button is set back to actual status.
Also the error handling is also done at the readback where you update the value_firePlaceModeActive then it is available as global and it is not emty anymore at the time where it is needed for the try except block. 
The last file works and I have already made a switch in my node-red automation. 
Only thing is that you can't press it very fast after and afer. The api call is very slow.
Please if you have time do a test with the updated file.
Best regards,
Hans

tuk90 commented:
Hi Hans,

If you remove the part in line 542 where you set the global variable again based on the value on the get calls it is fine. Rather I would see that you do it based on the turn on or of off the switch but he if it works its fine by me.

hansvanlin committed:
Update plugin.py

Removed the extra "get" as it was not needed.

hansvanlin commented:
Hi Nick, I removed the line 542.
